server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/HealthMetricsHandler.java

```diff
@@ -0,0 +1,217 @@
+         *       ],
+         *       "name" => "number-of-nodes",
+         *       "include-defaults" => true,
+         *       "resolve-expressions" => false,
+         *       "operation-headers" => {
+         *           "caller-type" => "user",
+         *           "access-mechanism" => "NATIVE"
+         *       }
+         *   }
+         */
+        final PathAddress address = PathAddress.pathAddress(operation.require(OP_ADDR));
+        final String cacheContainerName = address.getElement(CACHE_CONTAINER_INDEX).getValue();
+        final String metricName = operation.require(ModelDescriptionConstants.NAME).asString();
+        final ServiceController<?> controller = context.getServiceRegistry(false).getService(CacheContainerServiceName.CACHE_CONTAINER.getServiceName(cacheContainerName));
+        DefaultCacheContainer cacheManager = (DefaultCacheContainer) controller.getValue();
+
+        HealthMetrics metric = HealthMetrics.getMetric(metricName);
+        ModelNode result = new ModelNode();
+
+        if (metric == null) {
+            context.getFailureDescription().set(String.format("Unknown metric %s", metricName));
+        } else if (cacheManager == null) {
+            context.getFailureDescription().set(String.format("Unavailable cache container %s", metricName));
+        } else {
+            Health health = cacheManager.getHealth();
+            switch (metric) {
+                case CACHE_HEALTH:
+                    CacheHealth[] cacheHealths = health.getCacheHealth();
+                    List<String> perCacheHealth = new LinkedList<>();
+                    for (int i = 0; i < cacheHealths.length; ++i) {
+                        perCacheHealth.add(cacheHealths[i].getCacheName());
+                        perCacheHealth.add(cacheHealths[i].getStatus().toString());
+                    }
+                    result.set(toModelNodeCollection(perCacheHealth));
+                    break;
+                case FREE_MEMORY_KB:
+                    result.set(health.getHostInfo().getFreeMemoryInKb());
+                    break;
+                case TOTAL_MEMORY_KB:
+                    result.set(health.getHostInfo().getTotalMemoryKb());
+                    break;
+                case NUMBER_OF_NODES:
+                    result.set(health.getClusterHealth().getNumberOfNodes());
+                    break;
+                case CLUSTER_NAME:
+                    result.set(health.getClusterHealth().getClusterName());
+                    break;
+                case NUMBER_OF_CPUS:
+                    result.set(health.getHostInfo().getNumberOfCpus());
+                    break;
+                case CLUSTER_HEALTH:
+                    result.set(health.getClusterHealth().getHealthStatus().toString());
+                    break;
+                case LOG_TAIL:
+                    File path = new File(pathManager.resolveRelativePathEntry("server.log", ServerEnvironment.SERVER_LOG_DIR));
```

Comment:
This would only work if the user has not modified the default name / path for the server logs.